### PR TITLE
LPD-31536 Change initial state mechanism

### PR DESCRIPTION
**Pull request by DiegoHu97**
LPD-31536 Change initial state mechanism

# What is this trying to solve?

Change initial state mechanism of undo/redo component

[Link to ticket](https://liferay.atlassian.net/browse/LPD-31536)

# How am I fixing it?

As we were retrieving the initial state with a time out that waits for the page to be loaded, this was a very innefficient approach and caused some flaky tests. We have decided to send the initial state vaiues from metadata fields from the backend so in this way we don't have to wait until this values are loaded.

# How can I test it?

Execute the test journalArticle.spec.ts > 'LPD-26863: Undo/Redo buttons work with metadata fields' and expect to pass the test.

# Steps of the test:

- Enable FF 'LPD-11228': true,
		'LPD-15596': true,
- Create a WC
- Fill the title 
- Click on undo button
- See the title input value is empty string
- Click on redo button
- See the title appears in title input

Without this change this test does not pass because we had a time out of 1 second to wait for the page to load the values, and Playwright was filling the title faster than this timeout so teh undo/redo was not working for this initial state case.

### CODE REVIEW

**Comment by adolfopa** [comment]
ci:test:playwright

**Comment by DiegoHu97** [comment]
ci:stop

**Comment by DiegoHu97** [comment]
ci:test:playwright

File: modules/test/playwright/tests/journal-web/journalArticle.spec.ts [hunk 1]
```diff
@@ -255,13 +257,18 @@ autoSaveAsDraftTest(
...
+			await fillAndClickOutside(
+				page,
+				journalEditArticlePage.titleInput,
+				title
+			);
 
-		await expect(changesSavedIndicator).toBeVisible();
+			await expect(changesSavedIndicator).toBeVisible();
+
+			await expect(undoButton).toBeEnabled();
```
Comment (boton) [inline modules/test/playwright/tests/journal-web/journalArticle.spec.ts:270]: Can we only check for one thing inside `expect....toPass()`?

```js
await expect(async () => {
  // only one expect here
}).toPass();
```

I suspected that only `await expect(undoButton).toBeEnabled();` is needed.

File: modules/test/playwright/tests/journal-web/journalArticle.spec.ts [hunk 2]
```diff
@@ -198,23 +198,25 @@ autoSaveAsDraftTest(
...
+				journalEditArticlePage.titleInput,
+				title
+			);
 
-		await expect(changesSavedIndicator).toBeVisible();
+			await expect(changesSavedIndicator).toBeVisible();
 
-		await page.locator('body').click();
+			await expect(undoButton).toBeEnabled();
+		}).toPass();
```
Comment (boton) [inline modules/test/playwright/tests/journal-web/journalArticle.spec.ts:215]: Same

> Can we only check for one thing inside expect....toPass()?

**Comment by DiegoHu97** [comment]
ci:test:sf

**Comment by DiegoHu97** [comment]
ci:test:playwright

**Comment by DiegoHu97** [comment]
Hi @boton I just made the changes and some improvements, thanks for the review :heart:

**Review by boton — APPROVED** [review]
Looks awesome with these improvements 😄

**Comment by boton** [comment]
Let's wait until the `ci:test:playwright` is finished ⏳

**Comment by DiegoHu97** [comment]
ci:report:38388553

**Comment by DiegoHu97** [comment]
The failures are not related, moving forward the pr

**Comment by DiegoHu97** [comment]
ci:forward

**Pull request closed without merging**